zipline/finance/performance/position.py

```diff
@@ -65,22 +65,24 @@ def earn_dividend(self, dividend):
         Register the number of shares we held at this dividend's ex date so
         that we can pay out the correct amount on the dividend's pay date.
         """
-        assert dividend['sid'] == self.sid
-        out = {'id': dividend['id']}
+        out = {}
+
+        out['amount'] = self.amount * dividend.amount
+        return out
+
+    def earn_stock_dividend(self, stock_dividend):
+        """
+        Register the number of shares we held at this dividend's ex date so
```

Comment:
The return is the still number of shares, it is using the `position.amount`. Reading through there is collision with the dividend cash amount. Maybe we change the name of the filed in dividends to be `cash_amount` to reduce confusion?